### PR TITLE
COIOS-504: Added Atome Component Implementation

### PR DESCRIPTION
**Pull request by neelSharma12**
This PR contains Atome Component Implementation below is the current UI screenshot: 
AtomeComponentUI

Things pending: 
1. I will raise separate PR for UI changes related to billing address. 
2. Currently, this PR contains only unit test cases. I will be raising separate PR for to separate the Unit tests from  UI tests by creating separate target in the SDK.

### CODE REVIEW

**Review by mohammedDehairy — DISMISSED** [review]

**Review by nauaros — DISMISSED** [review]

File: AdyenComponents/Atome/AtomeComponent.swift [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        localizedString(.continueTitle, configuration.localizationParameters)
+    }
+
+    /// :nodoc:
+    override public func createPaymentDetails() -> PaymentMethodDetails {
+        guard let firstName = firstNameItem?.value,
+              let lastName = lastNameItem?.value,
+              let telephoneNumber = phoneItem?.value,
+              let billingAddress = addressItem?.value else {
+            fatalError("There seems to be an error in the BasicPersonalInfoFormComponent configuration.")
```
Comment (kaphacius) [inline AdyenComponents/Atome/AtomeComponent.swift]: maybe it's better to have this as an `assertionFailure`, rather than `fatalError`

**Review by kaphacius — DISMISSED** [review]
Should we have the `Internal` submodule tracked by git?

File: AdyenComponents/Atome/AtomeComponent.swift (hunk 1 above)
Comment (neelSharma12) [reply AdyenComponents/Atome/AtomeComponent.swift]: @kaphacius assertionFailure() do not guarantee that execution will stop if the assert is triggered and I guess we would like to stop the execution as all this information is required for the payment. Please let me know if I am wrong here.

Comment (mohammedDehairy) [reply AdyenComponents/Atome/AtomeComponent.swift]: we should't crash the merchant app, so I am afraid Yurii has a point, `assertionFailure` makes more sense IMO

Comment (neelSharma12) [reply AdyenComponents/Atome/AtomeComponent.swift]: Okay then I will change this to `assertionFailure`. Thanks for pointing this out.

Comment (descorp) [reply AdyenComponents/Atome/AtomeComponent.swift]: we have `Adyen. assertionFailure()` for this.
But I would make this check on `didSubmit` so we can return it to shopper as validation error

Comment (neelSharma12) [reply AdyenComponents/Atome/AtomeComponent.swift]: I have modified the code as per @descorp suggestion.

**Review by nauaros — APPROVED** [review]

**Review by mohammedDehairy — APPROVED** [review]

**Review by erenbesel — APPROVED** [review]

File: AdyenComponents/Atome/AtomeComponent.swift [hunk 2]
```diff
@@ -68,12 +68,12 @@ public final class AtomeComponent: AbstractPersonalInformationComponent {
...
 
     /// :nodoc:
-    override public func createPaymentDetails() -> PaymentMethodDetails {
+    override public func createPaymentDetails() throws -> PaymentMethodDetails {
         guard let firstName = firstNameItem?.value,
               let lastName = lastNameItem?.value,
               let telephoneNumber = phoneItem?.value,
               let billingAddress = addressItem?.value else {
-            fatalError("There seems to be an error in the BasicPersonalInfoFormComponent configuration.")
+                  throw UnknownError(errorDescription: "There seems to be an error in the BasicPersonalInfoFormComponent configuration")
```
Comment (descorp) [inline AdyenComponents/Atome/AtomeComponent.swift:76]: In another PR we can apply same behaviour for other `AbstractPersonalInformationComponent` based components..

**Review by descorp — APPROVED** [review]

File: AdyenComponents/Atome/AtomeComponent.swift (hunk 2 above)
Comment (neelSharma12) [reply AdyenComponents/Atome/AtomeComponent.swift:76]: sure.

**Pull request merged**